iceoryx_hoofs/include/iceoryx_hoofs/memory/relative_pointer.inl

```diff
@@ -99,17 +98,17 @@ inline typename std::enable_if<!std::is_void<U>::value, const U&>::type Relative
 }
 
 template <typename T>
-inline T* RelativePointer<T>::operator->() const noexcept
+inline typename RelativePointer<T>::ptr_t RelativePointer<T>::operator->() const noexcept
 {
-    auto* ptr = get();
+    auto* const ptr{get()};
```

Comment:
The `const` is also questionable, only more text to express the same thing (bad, to me). The pointer is obviously not changed and only returned. There is just no need to do this.

Maybe Axivion requires it but ideally we can do without IMO.

Reply:
This is a finding of Axivion using the AUTOSAR ruleset.